src/internal.h

```diff
@@ -928,7 +928,7 @@ _dispatch_ktrace_impl(uint32_t code, uint64_t a, uint64_t b,
 		dispatch_assert(_length != -1); \
 		_msg = (char *)malloc((unsigned)_length + 1); \
 		dispatch_assert(_msg); \
-		_snprintf(_msg, (unsigned)_length, "%s" fmt, DISPATCH_ASSERTION_FAILED_MESSAGE, ##__VA_ARGS__); \
+		snprintf(_msg, (unsigned)_length + 1, "%s" fmt, DISPATCH_ASSERTION_FAILED_MESSAGE, ##__VA_ARGS__); \
```

Comment:
Bleh; too bad that we still have the `_scprintf` here.  Can we use `snprintf` with a 0 size instead to get the desired length?  Then we could just use this version globally rather than having a windows only path.  The difference between the `asprintf` version and this would be the extra calls, but it is on the abort path anyways, so I don't think that it matters too much.

Reply:
Yeah, I agree this is kinda nasty - is it fine if I make that change as a follow-up?

asprintf is also used in several places in test code :(